presto-feature-toggle/src/main/java/com/facebook/presto/features/config/ConfigurationParser.java

```diff
@@ -0,0 +1,140 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.features.config;
+
+import com.facebook.airlift.json.JsonObjectMapperProvider;
+import com.facebook.presto.features.strategy.FeatureToggleStrategyConfig;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.isReadable;
+
+public class ConfigurationParser
+{
+    private static final String JSON = "JSON";
+    private static final String PROPERTIES = "PROPERTIES";
+    private static final String FEATURE = "feature";
+    private static final String FEATURE_S = "feature.%s.";
+    private static final String ENABLED = "enabled";
+    private static final String HOT_RELOADABLE = "hot-reloadable";
+    private static final String TRUE = "true";
+    private static final String FALSE = "false";
+    private static final String FEATURE_CLASS = "featureClass";
+    private static final String FEATURE_INSTANCES = "featureInstances";
+    private static final String CURRENT_INSTANCE = "currentInstance";
+    private static final String DEFAULT_INSTANCE = "defaultInstance";
+    private static final String STRATEGY = "strategy";
+    private static final String STRATEGY_DOT = "strategy.";
+    private static final String REGEX_DOT = "\\.";
+    private static final String EMPTY_STRING = "";
+    private static final String COMMA = ",";
+
+    private ConfigurationParser() {}
+
+    public static Map<String, FeatureConfiguration> parseConfiguration(FeatureToggleConfig config)
+    {
+        Map<String, FeatureConfiguration> featureConfigurationMap = new ConcurrentHashMap<>();
+        Path path = Paths.get(config.getConfigSource());
+        if (!path.isAbsolute()) {
+            path = path.toAbsolutePath();
+        }
+        checkArgument(exists(path), "File does not exist: %s", path);
+        checkArgument(isReadable(path), "File is not readable: %s", path);
+        if (JSON.equalsIgnoreCase(config.getConfigType())) {
+            parseJson(path, featureConfigurationMap);
+        }
+        else if (PROPERTIES.equalsIgnoreCase(config.getConfigType())) {
+            parseProperties(path, featureConfigurationMap);
+        }
+        return featureConfigurationMap;
+    }
+
+    private static void parseProperties(Path path, Map<String, FeatureConfiguration> featureConfigurationMap)
```

Comment:
Please parse properties consistently with Airlift.  We should not implement a new properties file parser for feature toggles.

Reply:
During start-up, Airlift parses configurations into pre-defined classes. To enable updates in configuration, it is necessary to implement a custom feature toggle properties parser.